### PR TITLE
Fixed incorrect diff for `num_clusters` and `min_num_clusters` for stopped `databricks_sql_endpoint`

### LINKED ISSUE #1204: [ISSUE] Incorrect diff for num_clusters in SQL Endpoint when endpoint is not running
Hi,

### Configuration
```hcl
 resource "databricks_sql_endpoint" "this" {
   name             = "Terraform-testing Endpoint"
   cluster_size     = "Large"
   min_num_clusters = 1
   max_num_clusters = 2
   auto_stop_mins   = 10
 }
```

### Expected Behavior
No change on subsequent applies when endpoint is not running

### Actual Behavior
image

### Steps to reproduce
create the endpoint
**turn it off**
`terraform apply` again

### Terraform and provider versions
provider 0.5.4

### Notes
Already reported in https://github.com/databrickslabs/terraform-provider-databricks/issues/1116 with attached fix: https://github.com/databrickslabs/terraform-provider-databricks/pull/1117

however the fix does not seem to work

### PR DESCRIPTION
**Pull request by nfx**
Correctly suppress diff for `num_clusters` on warehouses that are not running, in case `num_clusters` is not specified.

Note for contributors: `tf:"suppress_diff"` and `tf:"default"` don't work well together.

Fixes #1204 #1743

### CODE REVIEW

File: sql/resource_sql_endpoint.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ type SQLEndpoint struct {
 	AutoStopMinutes         int             `json:"auto_stop_mins" tf:"optional,default:120"`
 	MinNumClusters          int             `json:"min_num_clusters,omitempty" tf:"default:1,suppress_diff"`
```
Comment (alexott) [inline sql/resource_sql_endpoint.go:28]: Should we do the same for `MinNumClusters` ?

**Review by alexott — CHANGES_REQUESTED** [review]
tests are failing :-)
plus we need to do the same for `MinNumClusers`.

Plus, maybe it makes sense to add `.AtLeastOneOf`  to `num_clusers` and `min_num_clusters`?

**Comment by nfx** [comment]
@alexott having a sound CI system makes me sometimes forget running tests locally. 

Added the same fix for `min_num_clusters`. not really willing to `. AtLeastOneOf` because the validation should happen on the backend.

**Review by alexott — APPROVED** [review]
lgtm

**Pull request merged**